### ISSUE TITLE
add validate_certs directive on unarchive module.

### ISSUE THREAD

**Issue by ArKam** [ansible/ansible-modules-core#5659]
##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME

unarchive
##### ANSIBLE VERSION

```
ansible 2.0.0.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```

##### OS / ENVIRONMENT

Ubuntu 16.04 LTS
##### SUMMARY

It would be nice to be able to allow a remote unarchive from self-signed or badly forged SSL HTTP Endpoint like it is possible with the validate_certs directive on the GET_URL module.
##### STEPS TO REPRODUCE

```

```
"N/A"

##### EXPECTED RESULTS

"N/A"

##### ACTUAL RESULTS

"N/A"

```

```

**Comment by ansibot** [comment]
@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
As I am not a big fan to port every possible flag from get_url to unarchive and I fundamentally disagree that modules become more complex because it's more convenient to use a single action. So if others want to implement this and merge this, I won't object but I won't do it myself.

needs_contributor

**Comment by ArKam** [comment]
That it's not a request for more complexity but rather to improve the way unarchive already handle SSL. That's a feature witch is mostely required in enterprise where you can't access outside repo but only internal trusted repo with SSL certificats that are provided by an internal CA. Ansible is mainly aiming to be use by enterprises.
It's a complete nonsens to have to use two steps (get_url + unarchive) only because of this issue.

**Comment by dagwieers** [comment]
You are entitled to an opinion, as am I.

And it's not just because of this issue that I have this opinion, but rather as the maintainer of the unarchive module I received many requests for all possible URL fetching functionality and options that mimic what **get_url** (or **uri**) already implement. (authentication, proxy support, SSL functionality and options, client certificates, other schemes like ftp, sftp, smb, ...)

I am not interested in an unarchive module that is excellent in fetching files if we already have fetch_url with that sole purpose. And I will use my personal time in making unarchive better at unarchiving files.

**Comment by bcoca** [comment]
@ArKam  please consider that @dagwieers is doing this as a volunteer, it is up to him what he wants to work on. 

That said, I mostly agree with him, the uri functionality is already fully supported in it's own module and copying all that functionality here is redundant. The increased complexity and maintenance is not worth the trouble to save 1 extra task from a playbook.

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29476
